test/Audio/AlResource.cpp

```diff
@@ -0,0 +1,17 @@
+#include <SFML/Audio/AlResource.hpp>
+
+#include <doctest/doctest.h>
+
+#include <type_traits>
+
+TEST_CASE("sf::AlResource class - [audio]")
+{
+    SUBCASE("Type traits")
+    {
+        static_assert(!std::is_constructible<sf::AlResource>::value);
+        static_assert(!std::is_copy_constructible<sf::AlResource>::value);
+        static_assert(std::is_copy_assignable<sf::AlResource>::value);
+        static_assert(!std::is_nothrow_move_constructible<sf::AlResource>::value);
+        static_assert(std::is_nothrow_move_assignable<sf::AlResource>::value);
+    }
+}
```

Comment:
```suggestion
#include <SFML/Audio/AlResource.hpp>

#include <type_traits>

// Compile-time tests.
static_assert(!std::is_constructible<sf::AlResource>::value);        static_assert(!std::is_copy_constructible<sf::AlResource>::value);        static_assert(std::is_copy_assignable<sf::AlResource>::value);        static_assert(!std::is_nothrow_move_constructible<sf::AlResource>::value);        static_assert(std::is_nothrow_move_assignable<sf::AlResource>::value);
```

There's really not any point in wrapping the `static_assert` calls into a `TEST_CASE` or `SUBCASE`, as they will be tested during compilation. The `SUBCASE` you have here is basically a compile-time no-op.

Reply:
Fair. That still accomplishes my goal of having a .cpp file where runtime tests may eventually be written but it means we no longer build the doctest header just to not even use it. I'll make that PR-wide refactor soon.